### PR TITLE
chore(comments): update comments module to typescript

### PR DESCRIPTION
**Pull request by cdriesler**
⚠️ IMO this PR is exempt from the latest "prioritize PRs over everything else" directive. Review at your leisure and I'll ping when it's getting stale/becomes urgent.
⚠️ The diffs are mad annoying here but I've done my best to keep them legible (`git mv` etc). The individual commits may be easier to check for specific file changes.

## Description & motivation

- Multi-region looms in the distance, with a mountain of refactor work to be done

## Changes:

- Moves comment module files to .ts
- Ignored test files and migrations. I'll get them too but they're they failed the pain-priority math

## To-do before merge:

- ~~How does `SmartTextEditorValueSchema` interact with the `string` typing on `CommentRecord`? Looks like a later change that .js didn't care about but .ts does? It appears we want `SmartTextEditorValueSchema` in all cases now, but there were just enough `string` references that I wasn't sure.~~
- ~~A few queries were not typed (at the gql level) to provide filter args. I defaulted their values to `false` because that's what the missing values (often written as `(...input.filter || {})`) would resolve to~~
- ~~Several resolvers did not strictly assert that optional/nullable values existed. I opted for non-null assertions here, even though they're icky, instead of throwing errors. This was to limit changes to business logic.~~

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/server/modules/comments/graph/resolvers/comments.ts [hunk 1]
```diff
@@ -13,70 +12,77 @@ const {
...
+import {
+  Resolvers,
+  ResourceIdentifier,
+  ResourceType
+} from '@/modules/core/graph/generated/graphql'
+import { GraphQLContext } from '@/modules/shared/helpers/typeHelper'
+import { CommentRecord } from '@/modules/comments/helpers/types'
+
+const getStreamComment = async (
+  { streamId, commentId }: { streamId: string; commentId: string },
```
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/server/modules/comments/graph/resolvers/comments.ts:70]: Yes, except that this is incremental work towards the multi-region refactor, and is as large as I could fit in the time I had between other things. Limited just to moving the files over to typescript with as little other changes as possible. The `domain` bits and stuff like this I expect to catch in pieces in the next swing.

If it's important to you that I make all these changes _now_ in all the resolvers here I'll do it Iain but I will be grumbling the whole time

File: packages/server/modules/comments/graph/resolvers/comments.ts [hunk 2]
```diff
@@ -230,14 +246,14 @@ module.exports = {
     async commentThreads(parent, args, context) {
       const stream = await context.loaders.commits.getCommitStream.load(parent.id)
       await authorizeProjectCommentsAccess({
-        projectId: stream.id,
+        projectId: stream!.id,
```
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/server/modules/comments/graph/resolvers/comments.ts:249]: As a rule, no. I find non-null assertions icky and avoid them. But this is already effectively what the javascript was doing. The goal of this PR was to refactor .js => .ts with as few logical changes as possible. If I guard against this in all cases, it's changes to several resolvers at once that haven't been touched in a while. Not sure what kind of subtle behavioral changes that might have and felt safer to `!` for now and handle them in another future change.

File: packages/server/modules/comments/graph/resolvers/comments.ts [hunk 3]
```diff
@@ -302,31 +317,31 @@ module.exports = {
         authCtx: ctx,
         requireProjectRole: true
       })
-      return await createCommentThreadAndNotify(args.input, ctx.userId)
+      return await createCommentThreadAndNotify(args.input, ctx.userId!)
```
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/server/modules/comments/graph/resolvers/comments.ts:320]: same as the other `!`

File: packages/server/modules/comments/index.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { moduleLogger } from '@/logging/logging'
```
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/server/modules/comments/index.ts:1]: I did `git mv` in [this commit](https://github.com/specklesystems/speckle-server/pull/2513/commits/83a1f9d0f955306ae0e867c869338efb0b25f770) and the changes to the file in [this commit](https://github.com/specklesystems/speckle-server/pull/2513/commits/55dc21a96438359ca2855635cad65fbda41443a0). I think this is just what we get? I found it a little easier to read if I had it ignore whitespace changes, at least?

File: packages/server/modules/comments/services/index.ts [hunk 5]
```diff
@@ -0,0 +1,397 @@
+import crs from 'crypto-random-string'
```
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/server/modules/comments/services/index.ts:1]: same as the other `git mv` comment

File: packages/server/modules/comments/graph/resolvers/comments.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/server/modules/comments/graph/resolvers/comments.ts:249]: v fair, done

**Comment by cdriesler** [comment]
@iainsproat multi-region mentioned again today so resurrecting this, TODO's are in order and in line with some style stuff we've decided on since

I do think the tests should be moved to ts as well but will catch that in a future PR?

cc @alemagio if you'd like to pick this up too

**Review by alemagio — APPROVED** [review]

**Review by alemagio — APPROVED** [review]

**Pull request merged**